### ISSUE TITLE
Rule Change: Enhance `prefer-object-has-own` to flag `Object.keys(obj).includes(key)`

### ISSUE THREAD

**Issue by omril1** [eslint/eslint#19443]
### What rule do you want to change?

Enhance `prefer-object-has-own` to flag `Object.keys(obj).includes(key)`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint prefer-object-has-own: "error" */

// ❌ Incorrect (should be flagged)
Object.keys(obj).includes(key);

// ✅ Correct (preferred)
Object.hasOwn(obj, key);
```

### What does the rule currently do for this code?

Ignored

### What will the rule do after it's changed?

It would warn and allow for an autofix to `Object.hasOwn`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

While the original rule was created for cleaner code, and this change aligns with, this proposal has performance benefits as well, it would help avoid unnecessary intermediate arrays and iterations.

**Comment by mdjermanovic** [comment]
Hi @omril1, thanks for the issue!

 > /* eslint prefer-object-has-own: "error" */
> 
> // ❌ Incorrect (should be flagged)
> Object.keys(obj).includes(key);
> 
> // ✅ Correct (preferred)
> Object.hasOwn(obj, key);

These two are not equivalent since the first checks only enumerable properties.

```js
const key = "foo";
const obj = Object.create(null, {
    [key]: {
        value: "bar",
        enumerable: false,
    }
});

console.log(Object.keys(obj).includes(key)); // false

console.log(Object.hasOwn(obj, key)); // true

```

Given that this rule only suggests a shorter alternative with the same functionality, I don't think this pattern should be included.

@eslint/eslint-team thoughts?

**Comment by omril1** [comment]
Sorry, I was under the impression they are both working on own enumerable, my bad.

**Comment by omril1** [comment]
I guess `Object.getOwnPropertyNames(obj).includes(key)` would be more correct, but I haven't seen it used in the wild, `Object.keys` is a lot more common

**Comment by nzakas** [comment]
I agree with @mdjermanovic, I don't think it makes sense to update the rule for this purpose. Right now, it gives the exact equivalent alternative. Anything else and we're starting to guess.
